### PR TITLE
Support SPIR-V DebugTypePointer

### LINKED ISSUE #4077: No debuginfo generated for buffer references
In HLSL, using the `vk::BufferPointer` syntax/, I don't see any debug info emitted for buffer pointer struct members when using the `-g3` options. There's no `OpDebugTypePointer` for the buffer reference and no `OpDebugTypeStruct` for the struct being pointed to, and indeed I don't see any code in slang to emit the `OpDebugTypePointer` instruction. With GLSL the `layout(buffer_reference)` syntax doesn't seem to supported at all.

### PR DESCRIPTION
**Pull request by jkwak-work**
Closes #4077

With this commit, a memory pointer type is always emitted.
When its base type is basic types, it will emit DebugTypeBasic and DebugTypePointer.
Otherwise, it will emit "uint64" as DebugTypeBasic.

### CODE REVIEW

File: source/slang/slang-ir-util.cpp [hunk 1]
```diff
@@ -230,8 +230,9 @@ bool isSimpleDataType(IRType* type)
         return true;
     case kIROp_ArrayType:
     case kIROp_UnsizedArrayType:
-    case kIROp_PtrType:
         return isSimpleDataType((IRType*)type->getOperand(0));
+    case kIROp_PtrType:
```
Comment (csyonghe) [inline source/slang/slang-ir-util.cpp:234]: I am not sure we need to change `isSimpleDataType` here.

The right fix is that if the value type of a pointer type is basic type, then we want to emit OpDebugTypePointer, plus the OpDebugValue logic to update the variable with a pointer typed value.

If the value type isn't basic, then we want to emit the pointer as uint64 (OpDebugTypeBasic), plus the OpDebugValue inst to update the variable with a uint64 value.

**Comment by jkwak-work** [comment]
I made some changes so that a memory pointer type is always emitted.
When its base type is basic types, it will emit DebugTypeBasic and DebugTypePointer.
Otherwise, it will emit "uintptr" as DebugTypeBasic.

I couldn't work on DebugValue because it currently doesn't seem to work.
I created [an new issue](https://github.com/shader-slang/slang/pull/4224) and it seems like the "storage class" is not set properly.

**Comment by csyonghe** [comment]
If you take one of our existing pointer test cases and turn on debug symbol, you should be able to exercise the debug value logic, right?

**Comment by jkwak-work** [comment]
Right. I saw one test case. I will try it.

**Comment by jkwak-work** [comment]
The test code had some problem and I fixed it.
It also tests DebugValue from the emitted SPIR-V asm.
"DebugValue" was already working fine.

**Review by csyonghe — APPROVED** [review]

**Pull request merged**